native/src/tablecloth.mli

```diff
@@ -728,34 +728,855 @@ module Char : sig
+
+    [squareRoot] returns [NaN] when its argument is negative. See {!Float.nan} for more.
+  *)
+
+  val square_root : t -> t
+
+  val log : t -> base:t -> t
+  (** Calculate the logarithm of a number with a given base.
+
+    {[Float.log ~base:10. 100. = 2.]}
+
+    {[Float.log ~base:2. 256. = 8.]} 
+  *)
+
+  val clamp : t -> lower:t -> upper:t -> t
+  (** Clamps [n] within the inclusive [lower] and [upper] bounds. 
+
+    {[Float.clamp ~lower:0. ~upper:8. 5. = 5.]}
+
+    {[Float.clamp ~lower:0. ~upper:8. 9. = 8.]}
+
+    {[Float.clamp ~lower:(-10.) ~upper:(-5.) 5. = -5.]}
+  *)
+
+
+  (** {1 Checks} *)
+
+  val isNaN : t -> bool
+  (** Determine whether a float is an undefined or unrepresentable number.    
+
+    {[Float.isNaN (0.0 / 0.0) = true]}
+
+    {[Float.(isNaN (squareRoot (-1.0)) = true]}
+
+    {[Float.isNaN (1.0 / 0.0) = false  (* Float.infinity {b is} a number *)]}
+
+    {[Float.isNaN 1. = false]}
+  *)
+
+  val is_nan : t -> bool
+
+  val isInfinite : t -> bool
+  (** Determine whether a float is positive or negative infinity.
+
+    {[Float.isInfinite (0. / 0.) = false]}
+
+    {[Float.(isInfinite (squareRoot (-1.)) = false]}
+
+    {[Float.isInfinite (1. / 0.) = true]}
+
+    {[Float.isInfinite 1. = false]}
+
+    {[Float.(isInfinite nan) = false]}
+
+    Notice that [NaN] is not infinite! 
+
+    For a [float] [n] to be finite implies that [Float.(not (isInfinite n || isNaN n))] evaluates to [true].
+  *)
+
+  val is_infinite : t -> bool
+
+  val maximum : t -> t -> t
+   (** Returns the larger of two [float]s, if both arguments are equal, returns the first argument
+
+    {[Float.maximum 7. 9. = 9.]}
+
+    {[Float.maximum (-4.) (-1.) = (-1.)]} *)
+
+  val minimum : t -> t -> t
+  (** Returns the smaller of two [float]s, if both arguments are equal, returns the first argument
+
+    {[Float.minimum 7.0 9.0 = 7.0]} 
+
+    {[Float.minimum (-4.0) (-1.0) = (-4.0)]} *)
+
+  val inRange : t -> lower:t -> upper:t -> bool
+  (** Checks if [n] is between [lower] and up to, but not including, [upper]. 
+    If [lower] is not specified, it's set to to [0.0]. 
+    
+    {[Float.inRange ~lower:2. ~upper:4. 3. = true]}
+  
+    {[Float.inRange ~lower:1. ~upper:2. 2. = false]}
+  
+    {[Float.inRange ~lower:5.2 ~upper:7.9 9.6 = false]}      
+  *)
+
+  val in_range : t -> lower:t -> upper:t -> bool
+
+  (** {1 Angles} *) 
+
+  val hypotenuse : t -> t -> t
+  (** [hypotenuse x y] returns the length of the hypotenuse of a right-angled triangle with sides of length [x] and [y], or, equivalently, the distance of the point [(x, y)] to [(0, 0)]. 
+
+    {[Float.hypotenuse 3. 4. = 5.]}
+  *)
+
+  val degrees : t -> t
+  (** Converts an angle in {{: https://en.wikipedia.org/wiki/Degree_(angle) } degrees} to {{: https://en.wikipedia.org/wiki/Radian } radians }.
+
+    {[Float.degrees 180. = 3.141592653589793]}
+  *)
+
+  val radians : t -> t
+  (** Convert radians to {{: https://en.wikipedia.org/wiki/Radian } radians }. 
```

Comment:
I dont understand this docstring. "Convert radians to radians"?

Reply:
The cos and sin functions expect to be called with a value in radians, this (and degrees) are intended (from my understanding of the elm docs) to be used as ‘labels’ e.g ‘cos (radians pi)’.

I didn’t think this was a great part of the elm docs and was considering proposing a radians module.

I will improve the docstring and give some examples of its intended usage.